proto/api.go

```diff
@@ -308,14 +307,139 @@ func (sr *ReverseScanResponse) Verify(req Request) error {
 	return nil
 }
 
+// GetInner returns the Request contained in the `oneof`.
+func (ru BatchRequest_RequestUnion) GetInner() Request {
```

Comment:
> If the `ru.Request` was simply the type we placed in the batch

What would its type be?

> I think it was implemented this way so that `ru.Request` could enforce that only certain types be assigned.

That's correct.

> The whole `BatchRequest_RequestUnion_foo` set of types seems unnecessary on the part of gogoprotobuf.

Semantics, but this was actually implemented in golang/protobuf, and then merged into gogo/protobuf.